packages/core-extensions/content-containers/exam-edit/src/edit/index.vue

```diff
@@ -0,0 +1,173 @@
+            @update="$emit('update:subcontainer', $event)"
+            @delete="$emit('delete:subcontainer', group, 'group')" />
+        </VExpansionPanels>
+        <VBtn
+          v-if="!disabled"
+          :disabled="!container.id"
+          color="primary-lighten-5"
+          class="my-5"
+          @click.stop="createGroup">
+          <VIcon class="pr-2">mdi-folder-plus-outline</VIcon>
+          Add Question Group
+        </VBtn>
+      </div>
+    </VExpandTransition>
+  </VCard>
+</template>
+
+<script lang="ts" setup>
+import { activity as activityUtils, numberToLetter } from '@tailor-cms/utils';
+import { computed, ref, watch } from 'vue';
+import type { Activity } from '@tailor-cms/interfaces/activity';
+import type { ContentElement } from '@tailor-cms/interfaces/content-element';
+import type { ContentElementCategory } from '@tailor-cms/interfaces/schema';
+import filter from 'lodash/filter';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import last from 'lodash/last';
+import pluralize from 'pluralize';
+
+import AssessmentGroup from './AssessmentGroup.vue';
+
+interface Props {
+  name: string;
+  container: Activity;
+  elements: Record<string, ContentElement>;
+  position: number;
+  activities: Record<string, Activity>;
+  embedElementConfig?: ContentElementCategory[];
+  contentElementConfig?: ContentElementCategory[];
+  config?: Record<string, any>;
+  disabled?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  embedElementConfig: () => [],
+  contentElementConfig: () => [],
+  config: () => ({}),
+  disabled: false,
+});
+const emit = defineEmits([
+  'delete',
+  'add:subcontainer',
+  'save:element',
+  'update:element',
+  'reorder:element',
+  'delete:element',
+  'update:subcontainer',
+  'delete:subcontainer',
+]);
+
+const expanded = ref<string>();
+const collapsed = ref(!!props.container.id);
+
+const groups = computed(() =>
+  filter(props.activities, { parentId: props.container.id }));
+
+const title = computed(() => `Exam ${numberToLetter(props.position)}`);
+const label = computed(() => {
+  const groupTotal = groups.value.length;
+  return `${groupTotal} ${pluralize('set', groupTotal)}`;
+});
+
+const examObjectives = computed(() => {
+  const activities = Object.values(props.activities);
+  const activity = find(activities, { id: props.container.parentId! });
+  const objectiveTypes = get(props.config, 'objectives') as string[];
+  if (!activity || !objectiveTypes) return [];
+  const children = activityUtils.getDescendants(activities, activity);
+  return filter(children, (it) => objectiveTypes.includes(it.type));
+});
+
+const createGroup = () => {
+  emit('add:subcontainer', {
+    type: 'ASSESSMENT_GROUP',
+    parentId: props.container.id,
+    position: groups.value.length + 1,
+  });
+};
+
+watch(() => groups.value.length, () => {
+  expanded.value = last(groups.value)?.uid;
+});
+</script>
+
+<style lang="scss" scoped>
+h3 {
```

Comment:
Consider eliminating h tag and use only styling as it depends on the extension injection point / prev doc hierarchy.